examples/stabilizer_code.py

```diff
@@ -0,0 +1,340 @@
+from typing import List, Tuple
+
+import numpy as np
+
+import cirq
+
+# Based on:
+# Stabilizer Codes and Quantum Error Correction
+# Daniel Gottesman
+# https://thesis.library.caltech.edu/2900/2/THESIS.pdf
+#
+# However, there were some mistakes in the thesis:
+# https://www2.perimeterinstitute.ca/personal/dgottesman/thesis-errata.html
+
+
+def _BuildByCode(mat: np.ndarray) -> List[str]:
+    """
+    Takes into input a matrix of Boolean interpreted as row-vectors, each having dimension 2 * n.
+    The matrix is converted into another matrix with as many rows, but this time the vectors
+    contain the letters I, X, Y, and Z representing Pauli operators.
+    """
+    out = []
+    n = mat.shape[1] // 2
+    for i in range(mat.shape[0]):
+        ps = ''
+        for j in range(n):
+            if mat[i, j] == 0 and mat[i, j + n] == 0:
+                ps += 'I'
+            elif mat[i, j] == 1 and mat[i, j + n] == 0:
+                ps += 'X'
+            elif mat[i, j] == 0 and mat[i, j + n] == 1:
+                ps += 'Z'
+            else:
+                ps += 'Y'
```

Comment:
Optional: The three conditionals can be simplified into

```python
k = 2*mat[i, j + n] + mat[i, j]
ps += "IXZY"[k]
```

Reply:
Done.